src/artifact.c

```diff
@@ -5687,15 +5687,13 @@ arti_invoke(obj)
 				int role_skill;
 				struct obj *pseudo;
 				coord cc;
+				pseudo = mksobj(SPE_FIREBALL, MKOBJ_NOINIT); // create here for the skill check
 
 				energy = toosoon ? 25 : 15;
-				pseudo = mksobj(SPE_FIREBALL, MKOBJ_NOINIT);
-				pseudo->blessed = pseudo->cursed = 0;
-				pseudo->quan = 20L;			/* do not let useup get it */
 				role_skill = max(P_SKILL(uwep_skill_type()), P_SKILL(spell_skilltype(pseudo->otyp)) );
```

Comment:
For example (about calling mksobj less), why not replace `pseudo->otyp` with `SPE_FIREBALL`?